libs/wingsdk/src/target-sim/queue.inflight.ts

```diff
@@ -58,13 +58,15 @@ export class Queue
     this.subscribers.push(s);
   }
 
-  public async push(message: string): Promise<void> {
-    // TODO: enforce maximum queue message size?
-    return this.context.withTrace({
-      message: `Push (message=${message}).`,
-      activity: async () => {
-        this.messages.push(new QueueMessage(this.retentionPeriod, message));
-      },
+  // TODO: enforce maximum queue message size?
+  public async push(...messages: string[]): Promise<void> {
+    messages.forEach(async (message) => {
+      await this.context.withTrace({
+        message: `Push (message=${message}).`,
+        activity: async () => {
+          this.messages.push(new QueueMessage(this.retentionPeriod, message));
+        },
+      });
     });
```

Comment:
By using `forEach` like this, I'm not if all of the promises will be awaited. For example in this simplified version, if you try running it, you'll see it prints "done" before "task1" or "task2".

```js
async function main() {
    let sleep = (ms) => new Promise(resolve => setTimeout(resolve, ms));
    let task1 = sleep(1000).then(() => console.log("task1"));
    let task2 = sleep(1000).then(() => console.log("task2"));
    let tasks = [task1, task2];
    tasks.forEach(async (task) => {
        await task;
    });
    console.log("done");
}
main();
```

Instead I think we need to use something like [Promise.all](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/all) or a regular for loop...

Reply:
Ok, I didn't know about the differences of how `forEach` handles promises internally. I tested both `Promise.all` and `for...loop` but I couldn't notice a significant performance difference. My bias was that the former one would be more efficient since it can await multiple promises in parallel and stop execution as soon as one rejects.

Reply:
Ok, I didn't know that `forEach()` cannot handle promises properly. I tested both `Promise.all` and `for...loop` but I couldn't notice a significant performance difference. I thought that the former one would be more efficient since it can await multiple promises in parallel and stop execution as soon as one rejects.